code/game/turfs/turf.dm

```diff
@@ -130,15 +135,15 @@
 			qdel(A)
 		return
 	REMOVE_FROM_SMOOTH_QUEUE(src)
-	// Adds the adjacent turfs to the current atmos processing
-	for(var/turf/simulated/T in atmos_adjacent_turfs)
-		SSair.add_to_active(T)
-	SSair.remove_from_active(src)
 	visibilityChanged()
 	QDEL_LIST_CONTENTS(blueprint_data)
 	initialized = FALSE
 	..()
 
+/turf/ChangeTurf(path, defer_change = FALSE, keep_icon = TRUE, ignore_air = FALSE, copy_existing_baseturf = TRUE)
+	. = ..()
+	set_tile_atmos(x, y, z, atmos_mode = atmos_mode, external_temperature = initial(temperature))
```

Comment:
```suggestion
/turf/ChangeTurf(path, defer_change = FALSE, keep_icon = TRUE, ignore_air = FALSE, copy_existing_baseturf = TRUE)
	. = ..()
	set_tile_atmos(x, y, z, atmos_mode = atmos_mode, external_temperature = initial(temperature))
```
This isn't an inbuild just slap this at the bottom of the original function

Reply:
Yeah, caught that independently while fixing bugs.